backend/alerts/integrations/microsoft-teams/message_templates.go

```diff
@@ -0,0 +1,349 @@
+	"type": "AdaptiveCard",
+	"version": "1.0",
+	"body": [
+		{
+			"type":   "TextBlock",
+			"size":   "Large",
+			"weight": "Bolder",
+			"text":   "{{.Title}}",
+			"spacing": "Small",
+			"horizontalAlignment":      "Left",
+			"verticalContentAlignment": "Center"
+		},
+		{
+			"type":   "TextBlock",
+			"text":   "{{.Description}}",
+			"wrap": true,
+			"spacing": "Small",
+			"horizontalAlignment":      "Left",
+			"verticalContentAlignment": "Center"
+		},
+		{
+			"type": "FactSet",
+			"facts": {{.Facts}}
+		}
+	],
+	"actions": [
+		{
+			"type":  "Action.OpenUrl",
+			"title": "{{.ActionTitle}}",
+			"url":   "{{.ActionURL}}"
+		}
+	]
+  }`)
+
+var TrackPropertiesTemplate = []byte(`{
+	"$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
+	"type": "AdaptiveCard",
+	"version": "1.0",
+	"body": [
+		{
+			"type":   "TextBlock",
+			"size":   "Large",
+			"weight": "Bolder",
+			"text":   "{{.Title}}"
+		},
+		{
+			"type":   "TextBlock",
+			"text":   "{{.Description}}"
+		},
+		{
+			"type": "ColumnSet",
+			"columns": [
+				{
+					"type": "Column",
+					"items": [
+						{
+							"type":   "TextBlock",
+							"size":   "Medium",
+							"weight": "Bolder",
+							"text":   "Matched Track Properties"
+						},
+						{
+							"type":   "TextBlock",
+							"text":   "{{.MatchedValues}}",
+							"wrap":    true,
+							"width":   "stretch",
+							"spacing": "Small"
+						}
+					]
+				}
+			]
+		}
+		{{ if .RelatedValues }}
+			,{
+				"type": "ColumnSet",
+				"columns": [
+					{
+						"type": "Column",
+						"items": [
+							{
+								"type":   "TextBlock",
+								"size":   "Large",
+								"weight": "Bolder",
+								"text":   "Matched Track Properties"
+							},
+							{
+								"type":   "TextBlock",
+								"text":   "{{.RelatedValues}}",
+								"wrap": true,
+								"width":                    "stretch",
+								"spacing":                  "Small"
+							}
+						]
+					}
+				]
+			}
+		{{end}}
+	]
+  }`)
+
+var UserPropertiesTemplate = []byte(`{
+	"$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
+	"type": "AdaptiveCard",
+	"version": "1.0",
+	"body": [
+		{
+			"type":   "TextBlock",
+			"size":   "Large",
+			"weight": "Bolder",
+			"text":   "{{.Title}}"
+		},
+		{
+			"type":   "TextBlock",
+			"text":   "{{.Description}}"
+		},
+		{
+			"type": "ColumnSet",
+			"columns": [
+				{
+					"type": "Column",
+					"items": [
+						{
+							"type":   "TextBlock",
+							"size":   "Large",
+							"weight": "Bolder",
+							"text":   "Matched Track Properties"
+						},
+						{
+							"type":   "TextBlock",
+							"text":   "{{.MatchedUserProperties}}",
+							"wrap":    true,
+							"width":   "stretch",
+							"spacing": "Small"
+						}
+					]
+				}
+			]
+		}
+	],
+	"actions": [
+		{
+			"type":  "Action.OpenUrl",
+			"title": "{{.ActionTitle}}",
+			"url":   "{{.ActionURL}}"
+		}
+	]
+}`)
+
+var ErrorAlertMessageTemplate = []byte(`{
```

Comment:
Most messages sent by the ms teams bot use adaptive cards.
Even though they need to be valid json, I treat them like templates.
The "rendered" template must however be valid json.